Sources/Layout/MessageSizeCalculator.swift

```diff
@@ -75,32 +76,14 @@ open class MessageSizeCalculator: CellSizeCalculator {
 
     open func cellContentHeight(for message: MessageType, at indexPath: IndexPath) -> CGFloat {
 
-        let avatarVerticalPosition = avatarPosition(for: message).vertical
-        let avatarHeight = avatarSize(for: message).height
         let messageContainerHeight = messageContainerSize(for: message).height
         let bottomLabelHeight = cellBottomLabelSize(for: message, at: indexPath).height
-        let topLabelHeight = cellTopLabelSize(for: message, at: indexPath).height
+        let cellTopLabelHeight = cellTopLabelSize(for: message, at: indexPath).height
+        let messageTopLabelHeight = messageTopLabelSize(for: message, at: indexPath).height
         let messageVerticalPadding = messageContainerPadding(for: message).vertical
 
-        var cellHeight: CGFloat = 0
-
-        switch avatarVerticalPosition {
-        case .cellTop:
-            cellHeight += max(avatarHeight, topLabelHeight)
-            cellHeight += bottomLabelHeight
-            cellHeight += messageContainerHeight
-            cellHeight += messageVerticalPadding
-        case .cellBottom:
-            cellHeight += max(avatarHeight, bottomLabelHeight)
-            cellHeight += topLabelHeight
-            cellHeight += messageContainerHeight
-            cellHeight += messageVerticalPadding
-        case .messageTop, .messageCenter, .messageBottom:
-            cellHeight += max(avatarHeight, messageContainerHeight)
-            cellHeight += messageVerticalPadding
-            cellHeight += topLabelHeight
-            cellHeight += bottomLabelHeight
-        }
+        let cellHeight: CGFloat = cellTopLabelHeight + messageTopLabelHeight
```

Comment:
I'm not sure this is an accurate way to measure the height. What happens if the avatar is larger than the `messageContainerView` in some situations? Do you want me to work on this?

Reply:
@SD10 , 👌you can enhance it, the reason why I ignore the height of `avatarView`, is if we consider its height, we can hardly layout the views, e.x., think about `messageCenter` case, if `avatarView`'s height is enough bigger, like 1000px, the `messageContainerHeight`'s height is `10`px, for the original way, the height of cell is bigger than 1000px, and then, let's think about how to layout the subviews of cell, like `messageContainerView`, we have `padding`, it would break anything.